src/solver/infeasible-problem-analysis/constraint-slack-analysis.cpp

```diff
@@ -53,9 +71,30 @@ void ConstraintSlackAnalysis::run(MPSolver* problem)
     }
 
     hasDetectedInfeasibilityCause_ = true;
+
+    sortSlackVariablesByValue();
+    trimSlackVariables();
+}
+
+void ConstraintSlackAnalysis::selectConstraintsToWatch(MPSolver* problem)
+{
+    std::vector<std::string> patterns;
+    std::for_each(watchedConstraintTypes_.begin(),
+                  watchedConstraintTypes_.end(),
+                  [&](auto& c) { patterns.push_back(c->regexId()); });
+    std::string constraint_name_pattern{boost::algorithm::join(patterns, "|")};
```

Comment:
i think regex construction should not be splitted into multiple files!
for the sake of clarity, either here or in "watched constraint" classes

Reply:
Due to introduction of a factory, this part was moved, and also simplified. 
See file **watched-constraints.h/.cpp** files.